include/pybind11/pytypes.h

```diff
@@ -113,6 +113,10 @@ class object_api : public pyobject_tag {
     /// See above (the only difference is that the key is provided as a string literal)
     str_attr_accessor attr(const char *key) const;
 
+    // attr_with_type_hint is implemented in cast.h:
```

Comment:
Could you please also add
```
obj_attr_accessor attr_with_type_hint(handle key) const;
```
I think it could become really annoying (to users) if this overload is missing.

(I don't think we need the `object &&` overload; that's just an optimization that'll probably never have a measurable impact.)